app/ts/background/windows/personalSign.ts

```diff
@@ -112,54 +113,86 @@ export async function craftPersonalSignPopupMessage(ethereumClientService: Ether
 	}
 	const namedParams = { param: originalParams.params[1], account: originalParams.params[0] }
 	const account = getAddressMetaData(namedParams.account, userAddressBook)
-
-	if (namedParams.param.primaryType === 'Permit') {
-		const parsed = EIP2612Message.parse(namedParams.param)
-		const token = await getTokenMetadata(ethereumClientService, parsed.domain.verifyingContract)
-		const owner = getAddressMetaData(parsed.message.owner, userAddressBook)
-		if (token.type === 'NFT') throw 'Attempted to perform Permit to an NFT'
+	
+	let parsed 
+	try {
+		parsed = PersonalSignRequestIdentifiedEIP712Message.parse(namedParams.param)
+	} catch {
+		// if we fail to parse the message, that means it's a message type we do not identify, let's just show it as a nonidentified EIP712 message
 		return {
 			method: 'popup_personal_sign_request',
 			data: {
 				originalParams,
 				...basicParams,
-				type: 'Permit' as const,
-				message: parsed,
+				type: 'EIP712' as const,
+				message: namedParams.param,
 				account,
-				addressBookEntries: {
-					owner,
-					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
-					verifyingContract: token,
-				},
-				...await getQuarrantineCodes(BigInt(parsed.domain.chainId), account, activeAddressWithMetadata, owner),
+				quarantine: false,
+				quarantineCodes: []
 			}
 		} as const
 	}
```

Comment:
Don't use try/catch blocks for flow control.  `safeParse` exists for exactly this scenario, where you aren't confident that parsing will succeed and you need to branch on success/failure.  If necessary (doesn't look like it is here), you can access the error details in `maybeParsed.message`, `maybeParsed.fullError`, and `maybeParsed.key` inside of the failure code path.
```suggestion
	const maybeParsed = PersonalSignRequestIdentifiedEIP712Message.safeParse(namedParams.param)
	if (!maybeParsed.success) {
		// if we fail to parse the message, that means it's a message type we do not identify, let's just show it as a nonidentified EIP712 message
		return {
			method: 'popup_personal_sign_request',
			data: {
				originalParams,
				...basicParams,
				type: 'EIP712' as const,
				message: namedParams.param,
				account,
				quarantine: false,
				quarantineCodes: []
			}
		} as const
	}
	const parsed = maybeParsed.value
```